File: src/Server/WebApp/ClientApp/app/meeting/viewmeeting.ts
//import { WorkItem } from './workitem';
//import { Meeting } from '../models/meeting'
//import { Volunteer } from './volunteer';
//import { Location } from '../models/location';

export class ViewMeeting {
    meetingInfo: MeetingInfo;
    topicNames: string[];
    speakerNames: string[];
    topicDiscussions: TopicDiscussion[];
}

export class MeetingInfo {
    name: string;
    date: string;
}

export class TopicDiscussion {
    name: string;               // change field name to "topicName".
    speakers: Contribution[];   // change field name to "contributions".
}

export class Contribution {
    name: string;              // change field name to "speaker".
    text: string;
}

File: src/Server/WebApp/Features/Meetings/ViewMeeting.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WebApp.Models
{
    public class ViewMeeting
    {
        public Meeting meeting { get; set; }
        public string[] topicNames { get; set; }
        public string[] speakerNames { get; set; }
        public Topicdiscussion[] topicDiscussions { get; set; }
    }

    //public class MeetingInfo
    //{
    //    public string name { get; set; }
    //    public string date { get; set; }
    //    public string xxx { get; set; }
    //}

    public class Topicdiscussion
    {
        public string name { get; set; }
        public Speaker[] speakers { get; set; }
    }

    public class Speaker
    {
        public string name { get; set; }
        public string text { get; set; }
    }


}
